eth_bloom/bloom.py

```diff
@@ -58,32 +68,32 @@ def __contains__(self, value):
             in get_bloom_bits(value)
         )
 
-    def __index__(self):
+    def __index__(self) -> int:
         return operator.index(self.value)
 
-    def _combine(self, other):
+    def _combine(self: T, other: Union[int, T]) -> T:
```

Comment:
cc @cburgdorf 

At first I was going to suggest updating this to:  `_combine(self, other: Union[int, "BloomFilter") -> "BloomFilter"` but then I got to thinking about subclasses.  Are these functionally equivalent?  It appears that as it is, it is more precise but I don't know if that's how it actually plays out...

Reply:
Updating that to `_combine(self, other: Union[int, BloomFilter) -> BloomFilter` gave a `Flake8` error -> `F821 undefined name 'BloomFilter'`. Hence had to go about this way

Reply:
>but then I got to thinking about subclasses. Are these functionally equivalent?

I don't think there is a difference in terms of subclasses between using a generic `T` with `bound='BloomFilter'` or just using `BloomFilter`. Both should work with subtypes of `BloomFilter` just fine. I actually think making this generic doesn't add any value here.

@Bhargavasomu 

>Updating that to `_combine(self, other: Union[int, BloomFilter) -> BloomFilter` gave a Flake8 error -> F821 undefined name 'BloomFilter'

Just put `"BloomFilter"` in quotes (similar to how you did it with `bound='BloomFilter'`. It's called a [Forward Reference](https://mypy.readthedocs.io/en/latest/kinds_of_types.html#class-name-forward-references) and is basically a shortcoming of typing support in Python as it is today. They are still perfectly valid in terms of mypy type checking but they are at ease with `flake` / `python` ;)